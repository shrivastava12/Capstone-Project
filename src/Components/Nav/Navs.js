import React from "react";
import "./Navbar.css";
import logo from '../../Images/logo.png';
import { Link, Redirect } from "react-router-dom";
function Navs() {
  // let handleClick = () =>{
  //     return <Redirect to='/login'/>
  // }
  return (
    <>
      <nav className="navbar bg-dark navbar-expand-lg navbar-dark sticky-top">
        <a className="navbar-brand" href="#">
          <img  src={logo} width={70} height={70}/>
        </a>
        <button
          className="navbar-toggler"
          type="button"
          data-toggle="collapse"
          data-target="#navbarTogglerDemo02"
          aria-controls="navbarTogglerDemo02"
          aria-expanded="false"
          aria-label="Toggle navigation"
        >
          <span className="navbar-toggler-icon"></span>
        </button>

        <div
          className="collapse navbar-collapse item_div"
          id="navbarTogglerDemo02"
        >
          <ul className="navbar-nav mx-auto">
            <li className="nav-item active">
              <Link className="nav-link" href="#" style={{ color: "white" }} to='/'>
                Home <span className="sr-only">(current)</span>
              </Link>
            </li>

            <li className="nav-item active">
              <Link
                className="nav-link"
                href="#"
                style={{ color: "white" }}
                to="/song"
              >
                Songs
              </Link>
            </li>
            <li className="nav-item active">
              <Link className="nav-link" href="#" style={{ color: "white" }} to='/playlist'>
                Playlist
              </Link>
            </li>
            <li className="nav-item active">
              <a className="nav-link " href="#" style={{ color: "white" }}>
                Contact Us
              </a>
            </li>
          </ul>
          <Link to="/login">
            <button className="btn btn-outline-light">Login/SignUp</button>
          </Link>
        </div>
      </nav>
    </>
  );
}

export default Navs;
