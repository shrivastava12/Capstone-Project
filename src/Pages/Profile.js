import React from "react";
import { connect } from "react-redux";
import user from "../Images/user.png";
function Profile({users}) {
    console.log(users)
  return (
    <div>
      <div className="container mt-4 mb-4">
        <h2 data-testid='welcome' style={{ fontSize: "45px", color: "green", fontWeight: "bold" }}>
          Welcome! to your profile
        </h2>
        <div className="card mt-3 p-2">
          <div className="card-body">
         
            <div className="">
              <div className="row">
                <div className="col-lg-6">
                  <img src={user} height={400} width='100%' />
                </div>
                <div className="col-lg-6">
                  <div className="list-group mt-4">
                    <button
                      data-testid='username'
                      type="button"
                      className="list-group-item list-group-item-action active"
                      aria-current="true"
                    >
                      {users.first_name + " " + users.last_name}
                    </button>
                    <button
                      data-testid='email'
                      type="button"
                      className="list-group-item list-group-item-action"
                    >
                      {users.email}
                    </button>
                    <button
                      data-testid='location'
                      type="button"
                      className="list-group-item list-group-item-action"
                    >
                     {users.location}
                    </button>
                    <button
                      data-testid='mobile'
                      type="button"
                      className="list-group-item list-group-item-action"
                    >
                     +91 {users.mobile_no}
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}
const mapStateToProps = (state) => ({
    users:state.auth.user,
})
export default connect(mapStateToProps,{})(Profile);
