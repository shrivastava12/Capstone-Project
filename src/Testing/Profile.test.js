import {createStore} from 'redux';
import {Provider} from 'react-redux';
import {render,fireEvent,cleanup} from '@testing-library/react';
import Navs from '../Components/Nav/Navs';
import Profile from '../Pages/Profile';

afterEach(cleanup);


const loginState =  {
    isAuthenticated:false,
    user:JSON.parse(localStorage.getItem('user')),
    error:""
};

function reducer(state=loginState,action){
    const {type,payload} = action;
    switch (type) {
           case 'REGISTER_SUCCESS':
            localStorage.setItem('user',JSON.stringify(payload));
            return{
                ...state,
                isAuthenticated:true,
                user:payload,
                error:""
            }
        case 'LOGIN_SUCCESS':
            localStorage.setItem('user',JSON.stringify(payload));
            return{
                ...state,
                isAuthenticated:true,
                user:payload,
                error:""
            }
        case 'LOGIN_FAIL':
            return{
                ...state,
                isAuthenticated:false,
                user:localStorage.removeItem('user'),
                error:payload
            }
        case 'LOGOUT_SUCCESS':

            return{
                ...state,
                isAuthenticated:false,
                user:localStorage.removeItem('user'),
                error:""
            }
        default:
            return state;
    }
}

function renderWithRedux(
    component,{initialState,store=createStore(reducer,initialState)} = {}
){
    return {
        ...render(<Provider store={store}>{component}</Provider>)
    };
}


it('Render username  text correctly',() => {
    const {getByTestId} = renderWithRedux(<Profile/>,{
        initialState:{auth:{
 isAuthenticated:true,
             user:{
                 email: "abc@gmail.com",
                first_name: "Pankaj",
                id: "1",
                last_name: "shrivastava",
                location: "Earth",
                mobile_no: "7479530699",
                password: "1234"
             },
    error:""
        }
           
        }
    });
    expect(getByTestId('username')).toHaveTextContent('Pankaj shrivastava')
})

it('Render email  text correctly',() => {
    const {getByTestId} = renderWithRedux(<Profile/>,{
        initialState:{auth:{
 isAuthenticated:true,
             user:{
                 email: "abc@gmail.com",
                first_name: "Pankaj",
                id: "1",
                last_name: "shrivastava",
                location: "Earth",
                mobile_no: "7479530699",
                password: "1234"
             },
    error:""
        }
           
        }
    });
    expect(getByTestId('email')).toHaveTextContent('abc@gmail.com')
})

it('Render mobile  text correctly',() => {
    const {getByTestId} = renderWithRedux(<Profile/>,{
        initialState:{auth:{
 isAuthenticated:true,
             user:{
                 email: "abc@gmail.com",
                first_name: "Pankaj",
                id: "1",
                last_name: "shrivastava",
                location: "Earth",
                mobile_no: "7479530699",
                password: "1234"
             },
    error:""
        }
           
        }
    });
    expect(getByTestId('mobile')).toHaveTextContent('+91 7479530699')
})


it('Render location text correctly',() => {
    const {getByTestId} = renderWithRedux(<Profile/>,{
        initialState:{auth:{
 isAuthenticated:true,
             user:{
                 email: "abc@gmail.com",
                first_name: "Pankaj",
                id: "1",
                last_name: "shrivastava",
                location: "Earth",
                mobile_no: "7479530699",
                password: "1234"
             },
    error:""
        }
           
        }
    });
    expect(getByTestId('location')).toHaveTextContent('Earth')
})

it('Creating SnapShot of Profile',() => {
    const tree = renderWithRedux(<Profile/>,{
        initialState:{auth:{
 isAuthenticated:true,
             user:{
                 email: "abc@gmail.com",
                first_name: "Pankaj",
                id: "1",
                last_name: "shrivastava",
                location: "Earth",
                mobile_no: "7479530699",
                password: "1234"
             },
    error:""
        }
           
        }
    })
    expect(tree).toMatchSnapshot();
})
