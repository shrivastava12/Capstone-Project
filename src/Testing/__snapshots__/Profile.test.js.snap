// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creating SnapShot of Profile 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="container mt-4 mb-4"
        >
          <h2
            data-testid="welcome"
            style="font-size: 45px; color: green; font-weight: bold;"
          >
            Welcome! to your profile
          </h2>
          <div
            class="card mt-3 p-2"
          >
            <div
              class="card-body"
            >
              <div
                class=""
              >
                <div
                  class="row"
                >
                  <div
                    class="col-lg-6"
                  >
                    <img
                      height="400"
                      src="user.png"
                      width="100%"
                    />
                  </div>
                  <div
                    class="col-lg-6"
                  >
                    <div
                      class="list-group mt-4"
                    >
                      <button
                        aria-current="true"
                        class="list-group-item list-group-item-action active"
                        data-testid="username"
                        type="button"
                      >
                        Pankaj shrivastava
                      </button>
                      <button
                        class="list-group-item list-group-item-action"
                        data-testid="email"
                        type="button"
                      >
                        abc@gmail.com
                      </button>
                      <button
                        class="list-group-item list-group-item-action"
                        data-testid="location"
                        type="button"
                      >
                        Earth
                      </button>
                      <button
                        class="list-group-item list-group-item-action"
                        data-testid="mobile"
                        type="button"
                      >
                        +91 
                        7479530699
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="container mt-4 mb-4"
      >
        <h2
          data-testid="welcome"
          style="font-size: 45px; color: green; font-weight: bold;"
        >
          Welcome! to your profile
        </h2>
        <div
          class="card mt-3 p-2"
        >
          <div
            class="card-body"
          >
            <div
              class=""
            >
              <div
                class="row"
              >
                <div
                  class="col-lg-6"
                >
                  <img
                    height="400"
                    src="user.png"
                    width="100%"
                  />
                </div>
                <div
                  class="col-lg-6"
                >
                  <div
                    class="list-group mt-4"
                  >
                    <button
                      aria-current="true"
                      class="list-group-item list-group-item-action active"
                      data-testid="username"
                      type="button"
                    >
                      Pankaj shrivastava
                    </button>
                    <button
                      class="list-group-item list-group-item-action"
                      data-testid="email"
                      type="button"
                    >
                      abc@gmail.com
                    </button>
                    <button
                      class="list-group-item list-group-item-action"
                      data-testid="location"
                      type="button"
                    >
                      Earth
                    </button>
                    <button
                      class="list-group-item list-group-item-action"
                      data-testid="mobile"
                      type="button"
                    >
                      +91 
                      7479530699
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
